cores/esp32/IPAddress.cpp

```diff
@@ -220,10 +112,13 @@ String IPAddress::toString4() const
 
 String IPAddress::toString6() const
 {
-    StreamString s;
-    s.reserve(40);
-    printTo(s);
-    return s;
+    char szRet[40];
+    snprintf(szRet, sizeof(szRet), "%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x",
+            _address.bytes[0], _address.bytes[1], _address.bytes[2], _address.bytes[3],
+            _address.bytes[4], _address.bytes[5], _address.bytes[6], _address.bytes[7],
+            _address.bytes[8], _address.bytes[9], _address.bytes[10], _address.bytes[11],
+            _address.bytes[12], _address.bytes[13], _address.bytes[14], _address.bytes[15]);
+    return String(szRet);
```

Comment:
This consumes a significant amount of code to output a non-standard format. Is there any significant use for this raw output of IPv6 (which don't even verifify that the addres if actually v6)?

Reply:
I do not disagree. But if you provide a good solution (that does not depend on lwip), it can be proposed and merged upstream, giving benefits to much wider audience :)

Reply:
I found it, mentioned here: https://github.com/espressif/arduino-esp32/pull/8907#discussion_r1403999007

Some implementation is here: https://github.com/arduino/ArduinoCore-API/blob/82a5055a0588976c8df8c1ff3d978f62d68410f3/api/IPAddress.cpp#L302-L365